src/rgw/rgw_rest_s3.cc

```diff
@@ -1413,6 +1428,16 @@ int RGWPutObj_ObjStore_S3::get_params()
     return -EINVAL;
   }
 
+  append = s->info.args.exists("append");
+  if (append) {
+    string pos_str = s->info.args.get("position");
```

Comment:
any reason not to support this in RGWPutObj_ObjStore_Swift also?

Reply:
I agree, and people have requested something like this for swift--but I think it's fair for that to be done separately, because it does add to the testing and doc effort

Reply:
em..I think swift could also support append object...